src/cargo/util/toml/mod.rs

```diff
@@ -807,6 +808,7 @@ pub struct TomlProject {
     authors: Option<Vec<String>>,
     build: Option<StringOrBool>,
     metabuild: Option<StringOrVec>,
+    target: Option<String>,
```

Comment:
We'll want to gate this with a nightly manifest feature if it's set to `Some` to avoid insta-stabilizing this as well.